.github/workflows/ci.yml

```
@@ -154,8 +165,46 @@ jobs:
       PYTHON_BINDINGS: OFF # ROS2-sourced Python crashes in Ubuntu22.04
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: ${{needs.commit-cache.outputs.cache_commit_sha}}
       - run: .github/workflows/sudo.sh ./install_dependencies.sh "${PYTHON_BINDINGS}" "${STATIC_ANALYSIS}" "${ENABLE_ROS_TESTS}"
       - run: .github/workflows/build.sh -DCMAKE_BUILD_TYPE="${{matrix.build_type}}" -DSTATIC_ANALYSIS="${STATIC_ANALYSIS}" -DPYTHON_BINDINGS="${PYTHON_BINDINGS}" -DENABLE_TESTS=ON -DENABLE_ROS_TESTS=${ENABLE_ROS_TESTS}
       - run: .github/workflows/test.sh
 
-  # Memory and thread sanitiser are not tested due to known issues of both types inside Fast-DDS. May be added in the future when eProsima fixes the issues.
+  test-congested-networks-simple-discovery:
+    name: Test working in congested networks (Simple Discovery)
+    needs: commit-cache
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        packets_loss: [25%, 50%]
+        network_delay: [250ms, 500ms]
+    env:
+      NETWORK_LOSS: ${{matrix.packets_loss}}
```

```suggestion
      PACKETS_LOSS: ${{matrix.packets_loss}}
```

And then in the .sh , use `PACKETS_LOSS` as well.